sideboard/lib/_threads.py

```diff
@@ -9,6 +12,26 @@
 from sideboard.lib import log, on_startup, on_shutdown
 
 
+# inject our own code at the start of every thread's start() method which sets the thread name via prctl().
+# Python thread names will now be shown in external system tools like 'top', '/proc', etc.
+def _thread_name_insert(self):
+    if self.name:
+        # linux doesn't allow thread names > 15 chars, and we ideally want to see the end of the name.
+        # attempt to shorten the name if we need to.
+        shorter_name = self.name if len(self.name) < 15 else self.name.replace('CP Server Thread', 'CPServ')
+        prctl.set_name(shorter_name)
+    threading.Thread._bootstrap_inner_original(self)
+if sys.version_info[0] == 3:
+    threading.Thread._bootstrap_inner_original = threading.Thread._bootstrap_inner
+    threading.Thread._bootstrap_inner = _thread_name_insert
+else:
+    threading.Thread._bootstrap_inner_original = threading.Thread._Thread__bootstrap
+    threading.Thread._Thread__bootstrap = _thread_name_insert
+
+# set the name of the main thread
+prctl.set_name('sideboard_main')
+
```

Comment:
notes: 
in python3, the inner function is named ```threading.Thread._bootstrap_inner```
in python2, the inner function is named ```threading.Thread._Thread__bootstrap```